client.go

```diff
@@ -978,6 +989,9 @@ func (c *TunaSessionClient) reconnect(remoteAddr string, sessionID []byte, i int
 	if len(newPubAddrs.Addrs) > i {
 		conn, err = c.dialTcpConn(ctx, remoteAddr, sessionID, pb.SessionType_TCP, i, newPubAddrs.Addrs[i], config)
 		if err == nil {
+			c.Lock()
+			c.sessionConns[sessKey][connID(i)] = conn
```

Comment:
Do we really need this? It should be set already in `dialTcpConn()` L957

Reply:
Yes, removed.